File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/common/get/GetObjectsOperation.java
/*
 * Copyright 2016 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.common.get;

import java.util.List;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.FetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.id.IObjectId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.operation.IOperation;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("as.dto.common.get.GetObjectsOperation")
public abstract class GetObjectsOperation<ID extends IObjectId, FETCH_OPTIONS extends FetchOptions<?>> implements IOperation
{

    private static final long serialVersionUID = 1L;

    private List<? extends ID> objectIds;

    private FETCH_OPTIONS fetchOptions;

    protected GetObjectsOperation()
    {
    }

    public GetObjectsOperation(List<? extends ID> objectIds, FETCH_OPTIONS fetchOptions)
    {
        this.objectIds = objectIds;
        this.fetchOptions = fetchOptions;
    }

    public List<? extends ID> getObjectIds()
    {
        return objectIds;
    }

    public FETCH_OPTIONS getFetchOptions()
    {
        return fetchOptions;
    }

    @Override
    public String getMessage()
    {
        return toString();
    }

    @Override
    public String toString()
    {
        return getClass().getSimpleName() + (getObjectIds() != null ? " " + getObjectIds().size() + " object(s)" : "");
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/common/get/GetObjectsOperation.js
/**
 * @author pkupczyk
 */
define([ "stjs", "as/dto/common/operation/IOperation" ], function(stjs, IOperation) {
	var GetObjectsOperation = function(objectIds, fetchOptions) {
		this.objectIds = objectIds;
		this.fetchOptions = fetchOptions;
	};
	stjs.extend(GetObjectsOperation, null, [ IOperation ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.common.get.GetObjectsOperation';
		prototype.objectIds = null;
		prototype.fetchOptions = null;
		prototype.getObjectIds = function() {
			return this.objectIds;
		};
		prototype.getFetchOptions = function() {
			return this.fetchOptions;
		};
		prototype.getMessage = function() {
			return "GetObjectsOperation";
		};
	}, {
		objectIds : {
			name : "List",
			arguments : [ "IObjectId" ]
		},
		fetchOptions : "FetchOptions"
	});
	return GetObjectsOperation;
})